src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2121,8 +2121,13 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 static_assert_no_msg(AreContiguous(NI_Sve_Compute16BitAddresses, NI_Sve_Compute32BitAddresses,
                                                    NI_Sve_Compute64BitAddresses, NI_Sve_Compute8BitAddresses));
 
-                GetEmitter()->emitInsSve_R_R_R_I(ins, EA_SCALABLE, targetReg, op1Reg, op2Reg,
-                                                 (intrin.id - NI_Sve_Compute16BitAddresses), opt,
+                //  imm     intrinsic
+                //  1       NI_Sve_Compute16BitAddresses
+                //  2       NI_Sve_Compute32BitAddresses
+                //  3       NI_Sve_Compute64BitAddresses
+                //  0       NI_Sve_Compute8BitAddresses
+                size_t imm = ((intrin.id - NI_Sve_Compute16BitAddresses) + 1) % 4;
```

Comment:
We shouldn't be doing this, at least not without some sort of assert/validation the value is correct. The lack of such validation is why the break got introduced and is extremely important to have given CI doesn't actually execute SVE tests today due to lack of hardware.

On xarch, we have an explicit `lookupIval` and corresponding flag to indicate when such an "implicit" immediate value exists and needs to be resolved (https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsic.h#L559).

We would ideally have the same here.

Reply:
> at least not without some sort of assert/validation the value is correct.

This means at a minimum we should have some:
```
#if DEBUG
switch (intrin.id)
{
    case NI_Sve_Compute16BitAddresses:
    {
        assert(imm == 1);
        break;
    }

    case NI_Sve_Compute32BitAddresses:
    {
        assert(imm == 2);
        break;
    }
    case NI_Sve_Compute64BitAddresses:
    {
        assert(imm == 3);
        break;
    }

    case NI_Sve_Compute8BitAddresses:
    {
        assert(imm == 0);
        break;
    }

    default:
    {
        unreached();
    }
}
#endif // DEBUG
```

But, I think this isn't really the best approach and having the general `lookupIval` helper that uses a switch is more than sufficient. It makes the code clearer, safer, and allows the underlying native compiler to optimize the code itself if that's appropriate.

This shouldn't be a perf bottleneck in anycase, it's a one off switch in the context of the entire IR compilation and `Compute*BitAddresses` is not going to be so frequent that it is problematic (especially not in comparison to every other switch and more general handling we have).

Reply:
> But, I think this isn't really the best approach and having the general lookupIval helper that uses a switch and returns the direct ival per case is more than sufficient.

done